python/.pre-commit-config.yaml

```diff
@@ -4,5 +4,6 @@ repos:
     files: ^python/
     hooks:
       - id: ruff
+        exclude: ^(integrations/.*)
```

Comment:
I was hoping to avoid excluding ruff for a particular folder since that implies we accept inferior code quality in parts of the repo.
Also I think the same can be achieved by simply not using pre-commit? If that is the case then please remove that exemption :)

Reply:
Yeah fair enough, but it's a lingering problem I don't expect people to update old (3m+) integrations, so they will inevitably fall foul of more recent pythonic developments. As pre-commit is default, new people to the project when they commit upstream to `integrations/` are going to see a load of complaints they're not at fault for. 

We certainly can fix all the legacy integrations, but that is going to take a little time 😃